DoSomething/odoyle-scripts
# Refactor broadcast load test script.

This is a huge PR. Sorry! 🙏 

# What does this PR do?
It refactors the previous implementation to make way for more atomic control of scenarios.
It also adds a Web Socket server to centralize provisioning of mobile numbers.
It adds the `userResponse` scenario to test real life traffic of a set of users responding to a broadcast.

There still needs some cleanup. I have left a few TODOs in the code to come back to.

# How to test
1. Make sure all `.env` variables are set correctly.
3. set `DS_GAMBIT_CONVERSATIONS_USE_TWILIO_TEST_CREDS` to `true` in Conversations Staging. (If testing against staging)
1. `npm i`
2. `node mobile-nums-server.js` - separate terminal window.
3. `node index.js -u 5 -n 10 -s statusCallback` - separate terminal window.
> From the **Getting Started** section of the README.md (The version included in the PR.)

You should be able to see the test run and get results on the load test.
